### ISSUE TITLE
[Feature] Use Formula result in Rollup

### ISSUE THREAD

**Issue by blaszta** [nocodb/nocodb#3014]
Use case:
I have Product table which has Width, Depth & Height, and create Volume column (function, W*D*H)
Then I have Box table, which has several Product inside. I want to create Total Volume column (Rollup of Volume column in Product table) but can't select/use it .
<img src="https://i.imgur.com/1HmR4lu.png"></img>
<img src="https://i.imgur.com/tc9i8of.png"></img>
<img src="https://i.imgur.com/ByuhUG3.png"></img>

Note: Rollup to sum Width, Depth, etc. works fine

This is the opposite of this issue : [https://github.com/nocodb/nocodb/issues/1331](url)

**Comment by PiotrKrzyzek** [comment]
I believe this is a duplicate of my issue: https://github.com/nocodb/nocodb/issues/1331

**Comment by dstala** [comment]
To be tracked under https://github.com/nocodb/nocodb/issues/1331

**Comment by delormev** [comment]
I agree with OP that this is not the same as issue #1331. Formulas still can't be used in rollup fields. 

As another example:

Table A: items
- Item Name
- Item description

Table B: items ordered
- Customer
- Item (linked to record from A)
- Quantity ordered

Going back to Table A and adding a rollup field that calculates the sum of quantity ordered works. 
But if you add a formula in table B (eg. {Quantity ordered} - 1), it won't let me use it in a rollup field. 

Similar problem for formulas working with strings: adding a rollup field that takes the max(Customer) works. 
But if you add a formula in table B (eg. CONCAT("Mr. ", {Customer}), it won't let me use it in a rollup field. 

In both instances, if you use a Lookup instead of a rollup, the field shows but the result shows an Array for the formula, rather than the nicely formatted list for the original field.

**Comment by PiotrKrzyzek** [comment]
I concure with delormev: I was initially wrong and this is the 'inverse' of my original feature (bug?) request. This us about using a formula in a roll-up whereas mine was using a roll-up in a formula. 👍 

Both features together would be very powerful!

**Comment by delormev** [comment]
Is it a problem that it's marked as Closed in the Issue Board? Apologies for all the messages on this, but I was hoping to confirm it's on the backlog -- it's one of the handful of issues that is preventing me from migrating from AirTable!

**Comment by wingkwong** [comment]
Mark it open back. Currently virtual columns are not supported in rollup selection.

**Comment by dawidszpilewski** [comment]
This feature in my opinion is a Game Changer - I love NocoDB but most of my projects (actually all) need to use Rollups with Formulas Colums. At this moment I wait and believe that will be introduced soon, if not (I know there are a lot of other important issues) I will have to move to other alternative project - I just have no choice - can't wait. So it would be (for me, but I think for others also) very usefull to know if this feature is comming soon...

**Comment by nalafod** [comment]
> This feature in my opinion is a Game Changer - I love NocoDB but most of my projects (actually all) need to use Rollups with Formulas Colums. At this moment I wait and believe that will be introduced soon, if not (I know there are a lot of other important issues) I will have to move to other alternative project - I just have no choice - can't wait. So it would be (for me, but I think for others also) very usefull to know if this feature is comming soon...

Indeed after setting up my first Nocodb setup. I have encountered the above mentioned lack of ability. It would be a really good feature as most eventually want to rollup sums or other formulas.

**Comment by lpadula** [comment]
Any update?

**Comment by uebenes** [comment]
I also think, that this is a **must have feature** - its needed very often, that you have to post process some columns like check-boxes and selectors into numeric representations and rollup it in another table.

This feature is working in **Airtable** and in **baserows**. I think, that NocoDB is currently the best open source alternative project for Airtable - but without this feature its useless for many projects.

**Comment by woodmin** [comment]
Just encountered the same issue as @uebenes when trying implement  the following: "post process some columns like check-boxes and selectors into numeric representations and rollup it in another table". Very common, but not possible in NocoDB now. Great software, but this is a game-stopper for now.

**Comment by Brausmith** [comment]
I'm trying to perform a few tasks that Baserow can do, and this is one of them.
If implemented, this feature would allow me to calculate some rolled up total costs.

**Comment by felipemeres** [comment]
I also would really appreciate having this feature

**Comment by bitsByB** [comment]
Yes, same here. This one feature is extremely valuable ( sorry for the duplicate earlier).

**Comment by pascalwhoop** [comment]
Is there any high level guidance on how this could be implemented? If it's feasible to do this as a drive-by contribution I could roll up my sleeves for a day or two and get this out, but would appreciate complexity feeling from maintainers.

**Comment by seanakabry** [comment]
With apologies for raising an old thread, this is a deal-breaker for me too. Examples of use cases: rolling up a formula field which calculates duration based on start time, stop time, and breaks, to sum time spent on a particular task/job; or rolling up a cost field which multiplies hours worked by an hourly rate.

**Comment by jeffhaskin** [comment]
> With apologies for raising an old thread, this is a deal-breaker for me too. Examples of use cases: rolling up a formula field which calculates duration based on start time, stop time, and breaks, to sum time spent on a particular task/job; or rolling up a cost field which multiplies hours worked by an hourly rate.

This is also my exact use case. One of the core functions I need. My impression is that using rollups to roll up formula fields from linked records is the primary thing people do with rollup fields.

**Comment by Jihad** [comment]
Adding my vote here as well. I'm used to having kind of filters for rollups from airtable. Or at least, Rolling up a "specific View" of a table, where it has applied certain filters (kinda ugly work around). But anything other than just rolling up ALL rows.

**Comment by alexfacciorusso** [comment]
Same here, really needed!

**Comment by dstala** [comment]
Use of formula field as rollup column is now supported https://github.com/nocodb/nocodb/pull/10393
